opal/mca/shmem/posix/shmem_posix_common_utils.c

```diff
@@ -60,22 +60,65 @@
 #include "shmem_posix.h"
 #include "shmem_posix_common_utils.h"
 
+static char const lookup_table[] = {
+    '+', '-', '0', '1', '2', '3', '4', '5',
+    '6', '7', '8', '9', 'a', 'b', 'c', 'd',
+    'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l',
+    'm', 'n', 'o', 'p', 'q', 'r', 's', 't',
+    'u', 'v', 'w', 'x', 'y', 'z', 'A', 'B',
+    'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J',
+    'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R',
+    'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'
+};
+
+static size_t prepare_name(char * posix_file_name_buff, size_t size) {
+
+    /* Get the process's pid and print in base-64. */
+    uint32_t const pid = (uint32_t)getpid();
```

Comment:
Why not using `ltoa` with radix 64 and the copy only the last 3 chars ?

Reply:
Thanks, I wasn't familiar with that function. But it doesn't look to be part of the C standard, and isn't available on many platforms (e.g. my Linux cluster lacks it). And for those platforms that do support it, it seems that a radix of 36 is the maximum legal value.

Reply:
I was not aware of the limitation to radix 36.